cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_adapters.go

```diff
@@ -324,6 +326,17 @@ func (s *Server) buildPkgInstall(installedPackageName, targetCluster, targetName
 	}
 	prereleases := prereleasesVersionSelection(s.pluginConfig.defaultPrereleasesVersionSelection)
 
+	versionSelection := &vendirversions.VersionSelectionSemver{
+		Constraints: versionConstraints,
+		Prereleases: prereleases,
+	}
+
+	// Ensure the selected version can be, actually installed to let the user know before installing
+	elegibleVersion, err := versions.HighestConstrainedVersion([]string{pkgVersion}, vendirversions.VersionSelection{Semver: versionSelection})
```

Comment:
This is how kapp-ctrl is actually doing that; which makes me wonder: should we use `vendir` from carvel instead of other `semver` dependencies?

Reply:
If it suits our needs, sure.